### DISCUSSION TITLE
Add JXL support.

### DISCUSSION THREAD

**Discussion by FerrumMaster** [LycheeOrg/Lychee#2718] [Ideas]
### Enhancement

Hi,

Please add support JXL using [libjxl](https://github.com/libjxl/libjxl) when doing import.

I have converted my library to JXL as it takes 50-60% less space versus JPG and Adobe, Apple and ofc Linux has native support of this file format.

**Comment by FerrumMaster** [comment]
FF 152 added default JXL support. Please consider adding it, the storage cost these days is painful, not mentioning the reduced traffic as the files are really smaller.

**Reply by ildyria** [reply]
Hi,

JXL is barely supported anywhere, and by barely I am being very generous. See: https://caniuse.com/jpegxl

So to support jxl we would pretty much need to allow import on jxl format but immediately decode it to a usable format for wide range of web browser to generate the medium/small/thumbs. :/

**Reply by FerrumMaster** [reply]
Please add preferences to leave the medium/small/thums also in jxl, no conversion, otherwise that defeats the idea. Chrome, Safari and FF now supports it natively, what's with the barely supported statement(like AI trained on old sources?).

 You can do exports in Adobe products, Windows has plugin in store, same applies to any Linux, MacOS has native support.

Try out it yourself, what is the data saving ratio doing it that way, remaining the same IQ. It is >50% + HDR support and it is totally free.

**Reply by ildyria** [reply]
> what's with the barely supported statement(like AI trained on old sources?

I literally gave you the link of caniuse which is the authority with regard to what is supported where.
https://caniuse.com/jpegxl
See at the bottom all the * that it needs flags.

Literally more sources cites this issue:
- https://www.smallpics.io/blog/what-is-jpeg-xl/ March 26, 2026
- https://www.photoformatlab.com/blog/jpeg-xl-chrome-2026-image-format-guide April 6, 2026

**Reply by FerrumMaster** [reply]
> > what's with the barely supported statement(like AI trained on old sources?
> 
> I literally gave you the link of caniuse which is the authority with regard to what is supported where. https://caniuse.com/jpegxl See at the bottom all the * that it needs flags.
> 
> Literally more sources cites this issue:
> 
>     * https://www.smallpics.io/blog/what-is-jpeg-xl/ March 26, 2026
> 
>     * https://www.photoformatlab.com/blog/jpeg-xl-chrome-2026-image-format-guide April 6, 2026

That's outdated info. That's why replied that way.

As of FF 152 it it is enabled by default.

**Comment by d7415** [comment]
> Firefox now offers *experimental* support for the new [JPEG XL image format](https://jpegxl.info/), which generally provides better compression than WebP, JPEG, PNG, and GIF and is designed to supersede them. *You can enable it from the Firefox Labs panel in Settings.*

https://www.firefox.com/en-US/firefox/152.0/releasenotes/
